src/compiler.ts

```diff
@@ -798,8 +798,8 @@ export class Compiler extends DiagnosticEmitter {
             }
             this.ensureModuleExport(instanceName, instance, prefix);
           }
-        } else if (functionPrototype.is(CommonFlags.GENERIC)) {
-          this.warning(
+        } else if (functionPrototype.is(CommonFlags.GENERIC) && this.options.pedantic) {
```

Comment:
Just a nit: Can you move the second check into the block, in case we are going to add an `else`? I think that's how this is checked elsewhere as well.